glyphs2fontir/src/erase_open_corners.rs

```diff
@@ -0,0 +1,524 @@
+            PathSeg::Cubic(cubic_bez) => cubic_bez.p2,
+        };
+        let next_point = match self.next {
+            PathSeg::Line(line) => line.end(),
+            PathSeg::Quad(quad_bez) => quad_bez.p1,
+            PathSeg::Cubic(cubic_bez) => cubic_bez.p1,
+        };
+
+        !(point_is_left_of_line(self.seg, prev_point)
+            || point_is_left_of_line(self.seg, next_point))
+    }
+
+    /// The intersection of the two path segments
+    ///
+    /// If this is an open corner, the intersection point will be the new corner.
+    fn intersection(&self) -> Option<Intersection> {
+        seg_seg_intersection(self.prev, self.next)
+    }
+}
+
+//https://github.com/googlefonts/glyphsLib/blob/74c63244fdbe/Lib/glyphsLib/filters/eraseOpenCorners.py#L14
+// 'left' from the perspective of an observer standing on line.p0 and lookign at line.p1?
+fn point_is_left_of_line(line: Line, point: Point) -> bool {
+    let Line { p0: a, p1: b } = line;
+    (b.x - a.x) * (point.y - a.y) - (b.y - a.y) * (point.x - a.x) >= 0.0
+}
+
+/// given a slice of path elements, determine which *segments* might be open corners.
+fn iter_erasure_candidates(
+    els: &[PathEl],
+    is_closed: bool,
+) -> impl Iterator<Item = ErasureCandidate> + use<'_> {
+    iter_segment_triplets(els, is_closed)
+        .enumerate()
+        .filter_map(|(ix, (prev, seg, next))| {
+            if let PathSeg::Line(seg) = seg {
+                Some(ErasureCandidate {
+                    ix,
+                    seg,
+                    prev,
+                    next,
+                })
+            } else {
+                None
+            }
+        })
+}
+
+/// Iterate over all the length-three windows of segments.
+///
+/// This will loop once, so that the last segment begins the last window, e.g.
+///
+/// For the segments `s0, s1, s2, s3` this will produce
+/// `(s0, s1, s2), (s1, s2, s3), (s2, s3, s0), and (s3, s0, s1)`
+fn iter_segment_triplets(
+    els: &[PathEl],
+    is_closed: bool,
+) -> impl Iterator<Item = (PathSeg, PathSeg, PathSeg)> + use<'_> {
+    const EMPTY_SEG: PathSeg = PathSeg::Line(Line {
+        p0: Point::ZERO,
+        p1: Point::ZERO,
+    });
+
+    // if the path is closed, we want to loop around to consider the first two segments
+    // following the last.
+    let n_loop_around = if is_closed { 2 } else { 0 };
+    let mut iter = kurbo::segments(els.iter().copied())
+        .chain(kurbo::segments(els.iter().copied()).take(n_loop_around))
+        .peekable();
+
+    // if this is none we will end iteration on the first call to `next` anyway,
+    // and the code is simpler without this being an option
+    let mut prev = iter.next().unwrap_or(EMPTY_SEG);
+    std::iter::from_fn(move || {
+        let seg = iter.next()?;
+        let next = iter.peek()?;
+        let result = (prev, seg, *next);
+        prev = seg;
+        Some(result)
+    })
+}
+
+#[derive(Clone, Debug)]
+struct Intersection {
```

Comment:
intersection point is obvious. `t` may be magical to some readers so I suggest a comment to clarify what t0/t1 are.